src/main/java/net/openhft/chronicle/core/util/NanoSampler.java

```diff
@@ -18,7 +18,21 @@
 
 package net.openhft.chronicle.core.util;
 
+import net.openhft.chronicle.core.annotation.SingleThreaded;
+
+@SingleThreaded
 @FunctionalInterface
 public interface NanoSampler {
+
+    /**
+     * Records a sample at the provided {@code nanos} time (corresponding to a time given in nanoseconds).
+     * <p>
+     * Provided nanos must be derived from a coherent and non-decreasing source such as {@link System#nanoTime()}
```

Comment:
JLBH uses System.nanoTime internally so I think this can only be called with System.nanoTime  output